FL/Fl_SVG_Image.H

```diff
@@ -147,14 +147,15 @@ private:
   float svg_scaling_(int W, int H);
   void rasterize_(int W, int H);
   virtual void cache_size_(int &width, int &height);
-  void init_(const char *filename, const unsigned char *filedata, const Fl_SVG_Image *copy_source,
-             size_t length);
+  void init_(const char *name, const unsigned char *filedata, size_t length);
   Fl_SVG_Image(const Fl_SVG_Image *source);
 public:
   /** Set this to \c false to allow image re-scaling that alters the image aspect ratio.
    Upon object creation, proportional is set to \c true, and the aspect ratio is kept constant.*/
   bool proportional;
-  Fl_SVG_Image(const char *filename, const char *svg_data = NULL, size_t length = 0);
+  Fl_SVG_Image(const char *filename);
+  Fl_SVG_Image(const char *sharedname, const char *svg_data);
+  Fl_SVG_Image(const char *sharedname, const uchar *svg_data, size_t length);
```

Comment:
We should IMHO be consistent WRT the type of image data. Is it char, unsigned char, or uchar. The latter needs a typedef (exists in our codebase) but should IMHO not be used for the public API. I'd prefer `const unsigned char *` for all image constructors.
Other image types should be checked and made consistent as well.

PS: Do we maybe need constructors for both signed and unsigned char, or do we rely on the compiler "converting" the type? Or would that risk compiler warnings?

Reply:
Ok, no problem.

Reply:
We *do* have signed char for text data and unsigned char plus length for compressed data. FLUID outputs it like that as well.